File: frontend/src/lib/api/audit.ts
import { get } from './client';

export interface AuditEntry {
	timestamp: string;
	action: string;
	actor: string;
	resource_type: string;
	resource_name: string;
	status: string;
	details: Record<string, unknown>;
	client_ip?: string;
}

export const auditApi = {
	getLogs: (limit = 100, action = '', actor = '') =>
		get<{ logs: AuditEntry[]; count: number }>(
			`/api/audit/logs?limit=${limit}&action=${encodeURIComponent(action)}&actor=${encodeURIComponent(actor)}`
		)
};

File: backend/internal/domain/audit.go
package domain

import "time"

type AuditEntry struct {
	Timestamp    string                 `json:"timestamp"`
	Action       string                 `json:"action"`
	Actor        string                 `json:"actor"`
	ResourceType string                 `json:"resource_type"`
	ResourceName string                 `json:"resource_name"`
	Status       string                 `json:"status"`
	Details      map[string]interface{} `json:"details"`
	ClientIP     string                 `json:"client_ip,omitempty"`
}

func NewAuditEntry(action, actor, resourceType, resourceName, status, clientIP string) AuditEntry {
	return AuditEntry{
		Timestamp:    time.Now().UTC().Format(time.RFC3339Nano),
		Action:       action,
		Actor:        actor,
		ResourceType: resourceType,
		ResourceName: resourceName,
		Status:       status,
		Details:      map[string]interface{}{},
		ClientIP:     clientIP,
	}
}
